company-tng.el

```diff
@@ -106,12 +106,15 @@ confirm the selection and finish the completion."
      (let ((ov (make-overlay (point) (point))))
        (setq company-tng--overlay ov)
        (overlay-put ov 'priority 2))
-     (advice-add 'company-select-next :before-until 'company-tng--allow-unselected)
-     (advice-add 'company-fill-propertize :filter-args 'company-tng--adjust-tooltip-highlight))
+     (setq company-selection-default nil)
+     (unless company-selection-changed
+       (setq company-selection company-selection-default))
+     (advice-add 'company-select-next :before-until 'company-tng--allow-unselected))
```

Comment:
Perhaps we should merge the `company-tng--allow-unselected` advice into the definition of `company-select-next`?

Reply:
I've merged a part of going to another selection from unselected state. For going back to unselected state, I left that in `company-tng`. Since that should be exclusive behavior of `company-tng`

Reply:
It would have allowed us to dispense with the advices altogether.

In any case, this doesn't look like a very important feature. Let's maybe ask the author. @nikital, do you ever use it in practice?

Reply:
I sometime do use this though

Reply:
@dgutov Personally, this feature is useful. Sometimes I would select the first candidate, ponder for a moment, unselect it and continue typing. Since there is no way in `tng` to cancel a completion once something is selected, this provides a nice way out (`C-g` would close the popup but leave the completion inserted).

Reply:
I think that `company-tng` needs this as they need to cancel completion once something has been selected.
Other backends don't do this and actually wait for user explicit completion completing action.